Mathlib/MeasureTheory/Measure/LevyProkhorovMetric.lean

```diff
@@ -395,13 +444,188 @@ lemma continuous_levyProkhorov_to_probabilityMeasure :
 
-end Levy_Prokhorov_comparison
+end Levy_Prokhorov_is_finer
+
+
+
+section Levy_Prokhorov_metrizes_convergence_in_distribution
+
+/-! ### On separable spaces the Lévy-Prokhorov distance metrizes convergence in distribution -/
+
+open BoundedContinuousFunction TopologicalSpace
+
+variable {ι : Type*} {Ω : Type*} [MeasurableSpace Ω]
+
+/-- In a separable pseudometric space, for any ε > 0 there exists a countable collection of
+disjoint Borel measurable subsets of diameter at most ε that cover the whole space. -/
+lemma SeparableSpace.exists_measurable_partition_diam_le
+    (X : Type*) [PseudoMetricSpace X] [SeparableSpace X]
+    [MeasurableSpace X] [OpensMeasurableSpace X] {ε : ℝ} (ε_pos : 0 < ε) :
+    ∃ (As : ℕ → Set X), (∀ n, MeasurableSet (As n)) ∧ (∀ n, Bornology.IsBounded (As n)) ∧
+        (∀ n, diam (As n) ≤ ε) ∧ (⋃ n, As n = univ) ∧
+        (Pairwise (fun (n m : ℕ) ↦ Disjoint (As n) (As m))) := by
+  by_cases X_emp : IsEmpty X
+  · refine ⟨fun _ ↦ ∅, fun _ ↦ MeasurableSet.empty, fun _ ↦ Bornology.isBounded_empty, ?_, ?_,
+            fun _ _ _ ↦ disjoint_of_subsingleton⟩
+    · intro n
+      simpa only [diam_empty] using LT.lt.le ε_pos
+    · simp only [iUnion_empty]
+      apply Eq.symm
+      simp only [univ_eq_empty_iff, X_emp]
+  rw [not_isEmpty_iff] at X_emp
+  obtain ⟨xs, xs_dense⟩ := exists_dense_seq X
+  have half_ε_pos : 0 < ε / 2 := half_pos ε_pos
+  set Bs := fun n ↦ Metric.ball (xs n) (ε/2)
+  set As := disjointed Bs
+  refine ⟨As, ?_, ?_, ?_, ?_, ?_⟩
+  · exact MeasurableSet.disjointed (fun n ↦ measurableSet_ball)
+  · exact fun n ↦ Bornology.IsBounded.subset isBounded_ball <| disjointed_subset Bs n
+  · intro n
+    apply (diam_mono (disjointed_subset Bs n) isBounded_ball).trans
+    convert diam_ball half_ε_pos.le
+    ring
+  · have aux : ⋃ n, Bs n = univ := by
+      convert DenseRange.iUnion_uniformity_ball xs_dense <| Metric.dist_mem_uniformity half_ε_pos
+      exact (ball_eq_ball' _ _).symm
+    simpa only [← aux] using iUnion_disjointed
+  · exact disjoint_disjointed Bs
+
+variable [PseudoMetricSpace Ω] [OpensMeasurableSpace Ω]
+
+lemma continuous_probabilityMeasure_toLevyProkhorov [SeparableSpace Ω] :
+    Continuous (ProbabilityMeasure.toLevyProkhorov (Ω := Ω)) := by
+  rw [continuous_iff_continuousAt]
+  intro P
+  rw [continuousAt_iff']
+  intro ε ε_pos
+  have liminf_open :=
+    @ProbabilityMeasure.le_liminf_measure_open_of_tendsto Ω _ (𝓝 P) _ _ _ _ P id tendsto_id
+  have third_ε_pos : 0 < ε / 3 := by linarith
+  have third_ε_pos' : 0 < ENNReal.ofReal (ε / 3) := ofReal_pos.mpr third_ε_pos
+  obtain ⟨Es, Es_mble, Es_bdd, Es_diam, Es_cover, Es_disjoint⟩ :=
+    SeparableSpace.exists_measurable_partition_diam_le Ω third_ε_pos
+  have Es_union_incr : Monotone (fun (n : ℕ) ↦ ⋃ i ∈ Iio n, Es i) :=
+    fun _ _ hnm ↦ biUnion_mono (Iio_subset_Iio_iff.mpr hnm) (fun _ _ ↦ le_rfl)
+  have exhaust : Tendsto (P.toMeasure ∘ fun n => ⋃ i, ⋃ (_ : i < n), Es i) atTop (𝓝 (P univ)) := by
```

Comment:
Perhaps this `have exhaust := ...` (together with a few things around it) would be reasonable to split off as a stand-alone lemma from the long proof. The right general lemma is the existing `tendsto_measure_iUnion`. But perhaps also a suitable special case of it deserves to be stand-alone, to facilitate applying it for example in cases where we have probability measures and partitions.